store/sqlstore/post_store.go

```diff
@@ -1281,6 +1294,22 @@ func (s *SqlPostStore) search(teamId string, userId string, params *model.Search
 	return list, nil
 }
 
+func removeMysqlStopWordsFromTerms(terms string) (string, error) {
+	stopWords := make([]string, len(searchlayer.MYSQL_STOP_WORDS))
+	copy(stopWords, searchlayer.MYSQL_STOP_WORDS)
+	for i := 0; i < len(stopWords); i++ {
+		stopWords[i] = fmt.Sprintf(`\b%s\b`, stopWords[i])
+	}
+	re, err := regexp.Compile(strings.Join(stopWords, "|"))
```

Comment:
Testing again, the \b splits also with the hyphen and for words like father-in-law is broken. I've updated [my test](https://play.golang.com/p/RRCflQTIs37) with both approaches